aiida/orm/implementation/sqlalchemy/querybuilder/main.py

```diff
@@ -366,16 +368,13 @@ def _build(self) -> Query:
                 raise ValueError(f'Unknown tag {tag!r} in filters, known: {list(self._tag_to_alias)}')
             self._query = self._query.filter(self.build_filters(alias, filter_specs))
 
-        ######################### PROJECTIONS ##########################
-        # first clear the entities in the case the first item in the
-        # path was not meant to be projected
-        # attribute of Query instance storing entities to project:
+        # PROJECTIONS ##########################
 
-        # Mapping between entities and the tag used/ given by user:
+        # Reset mapping of tag -> field -> projection_index
         self._tag_to_projected_fields = {}
 
-        projection_count = 0
-        QUERYBUILD_LOGGER.debug('self._data["project"]: %s', self._data['project'])
+        projection_count = 1
```

Comment:
was this is a hidden bug or is this really necessary due to the new sqlalchemy version? It especially seems weird since in the following code you do `projection_count - 1` a lot

Reply:
See above ☝️ the first "projection" is what the query is initialised with but not part of the QueryBuilder projections.
Happy to jump on Zoom to chat, if I'm not explaining it well enough in text lol